src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,65 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { Select } from '../../utils/Select';
+import './TodoFilter.scss';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  setFilterOption: (filterOption: Select) => void,
+  setQuery: (newQuery: string) => void,
+  query: string,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  setFilterOption = () => { },
+  setQuery = () => { },
+  query,
+}) => {
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => (
+    setFilterOption(event.target.value as Select)
+  );
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => (
+    setQuery(event.target.value)
+  );
+  const handleQueryClear = () => setQuery('');
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={handleSelectChange}
+          >
+            <option value={Select.All}>All</option>
+            <option value={Select.Active}>Active</option>
+            <option value={Select.Completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={handleQueryChange}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {!!query.length && (
```

Comment:
```suggestion
        {query && (
```
`query` is a string, and an empty string is a falsy value, so you can do like this, just check if the query exists.
https://developer.mozilla.org/en-US/docs/Glossary/Falsy